src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -2131,12 +2159,25 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
 
                         if (element is string elementAsString)
                         {
-                            if (string.IsNullOrEmpty(elementAsString))
+                            if (_whiteSpace)
                             {
-                                throw new ValidationMetadataException(
-                                    "ArgumentCollectionContainsEmpty",
-                                    null,
-                                    Metadata.ValidateNotNullOrEmptyCollectionFailure);
+                                if (string.IsNullOrWhiteSpace(elementAsString))
+                                {
+                                    throw new ValidationMetadataException(
+                                        "ArgumentCollectionContainsEmptyOrWhiteSpace",
+                                        null,
+                                        Metadata.ValidateNotNullOrWhiteSpaceCollectionFailure);
+                                }
+                            }
+                            else
+                            {
+                                if (string.IsNullOrEmpty(elementAsString))
```

Comment:
Same here. Please note that you will need to do additional adjustments besides my suggestions. Otherwise, it won't compile.
```suggestion
                            else if (string.IsNullOrEmpty(elementAsString))
```